Lombiq.EInvoiceValidator.Tests/UnitTests/ValidatorTests.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Jering.Javascript.NodeJS;
+using Lombiq.EInvoiceValidator.Extensions;
+using Lombiq.EInvoiceValidator.Helpers;
+using Lombiq.EInvoiceValidator.Services;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.DependencyInjection;
+using Shouldly;
+using System.IO;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Lombiq.EInvoiceValidator.Tests.UnitTests;
+
+public class ValidatorTests
+{
+    [Fact]
+    public async Task TestInvoiceValidationHelper()
+    {
+        var services = new ServiceCollection();
+        services.AddEInvoiceValidationServices();
+        var serviceProvider = services.BuildServiceProvider();
+
+        var nodeJsService = serviceProvider.GetRequiredService<INodeJSService>();
+        var memoryCache = serviceProvider.GetRequiredService<IMemoryCache>();
+        var eInvoiceXmlSchemaSet = serviceProvider.GetRequiredService<IEInvoiceXmlSchemaSet>();
+
+        var filePaths = Directory.GetFiles(Path.Combine("UnitTests", "SampleInvoices"), "*.xml", SearchOption.AllDirectories);
+
+        foreach (var filePath in filePaths)
+        {
+            using var streamReaderInner = new StreamReader(filePath);
+            var result = await InvoiceValidationHelper.ValidateInvoiceAsync(
+                streamReaderInner.BaseStream,
+                nodeJsService,
+                memoryCache,
+                eInvoiceXmlSchemaSet,
+                cancellationToken: TestContext.Current.CancellationToken);
+
+            if (filePath.Contains("failing"))
+            {
+                result.Successful.ShouldBeFalse();
+            }
+            else
+            {
+                result.Successful.ShouldBeTrue();
```

Comment:
This would be better as a data-driven test, generating as many test cases as there are files. Alternatively, the messages of these assertions should contain the file paths, so you can start debugging somewhere.